### ISSUE TITLE
Question - Is it possible to restrict by state or province

### ISSUE THREAD

**Issue by samcro1967** [porech/caddy-maxmind-geolocation#3]
Wondering if it is possible to get more granular than country and restrict by subdivision or metro.  I see that maxmind has a file of cities which includes the state or province as subdivision 1.  City might be too specific, but state or even metro might be more reasonable.  Subdivision does not appear to be unique, but metro_code does.  Maybe have all 3 and only allow or deny when all three are true from a client?

Might look something like this:

(geofilter) {
		@mygeofilter {
			  maxmind_geolocation {
			  db_path "/data/GeoLite2-City.mmdb"
			  allow_countries US
                          allow_subdivisions AK AL AR
                          allow_metro_code 747 630 670
			}
		}
}

**Comment by ale-rinaldi** [comment]
Hello,

sure, I didn't do this just because I never needed it. I will happily work on this in the next few days.

Thanks

**Comment by samcro1967** [comment]
Just checking back to see if you were able to make any progress on this.

**Comment by samcro1967** [comment]
@ale-rinaldi Any progress on this enhancement?

**Comment by ale-rinaldi** [comment]
Hello @samcro1967 ,

I completed the feature in this branch: https://github.com/porech/caddy-maxmind-geolocation/tree/3-match-subdivisions-and-country-codes

I tested up some cases and they seem to work just fine, but before merging and releasing, I'm asking you to do some tests on it and confirm me that it matches your use case and, most important, you see no regressions.

So here's how you can test it:
- Install xcaddy [following the official documentation](https://github.com/caddyserver/xcaddy)
- Build caddy for your system with the plugin from the branch:
```shell
xcaddy build --with github.com/porech/caddy-maxmind-geolocation@3-match-subdivisions-and-country-codes
```
This will create the `caddy` executable in your current directory
- Test the matchers using the new `allow_subdivisions`, `deny_subdivisions`, `allow_metro_codes` and `deny_metro_codes` directives, following the samples in the branch README
- Let me know how it goes

Thanks!

**Comment by samcro1967** [comment]
@ale-rinaldi Everything seem s to be working for me.  Below is what I tested.  Appreciate the update!

1.  Tested with the new version using the city data base and only allow_countries.  Worked as expected.
2.  Added in only allow_subdivisions.  Worked as expected.
3.  Added in only allow_metro_codes.  Worked as expected.
4.  Added both allow_subdivisions and allow_metro_codes.  Worked as expected.
5.  Added in two locations for allow_subdivisions and allow_metro_codes.  Worked as expected.
